Kitodo/src/main/java/org/kitodo/production/services/command/KitodoScriptService.java

```diff
@@ -181,6 +187,44 @@ private boolean executeScript(List<Process> processes, String script) throws Dat
         return true;
     }
 
+    private void deleteData(List<Process> processes, String script) {
+        String currentProcessTitle = null;
+        try {
+            script = script.replaceFirst("\\s*action:deleteData\\s+(.*?)[\r\n\\s]*", "$1");
+            DeleteDataScript deleteDataScript = new DeleteDataScript();
+            for (Process process : processes) {
+                currentProcessTitle = process.getTitle();
+                LegacyMetsModsDigitalDocumentHelper metadataFile = ServiceManager.getProcessService()
+                        .readMetadataFile(process);
+                deleteDataScript.process(metadataFile, process, script);
+                ServiceManager.getMetsService().saveWorkpiece(metadataFile.getWorkpiece(),
+                        ServiceManager.getProcessService().getMetadataFileUri(process));
+                Helper.setMessage("deleteDataOk", currentProcessTitle);
+            }
+        } catch (IOException e) {
+            Helper.setErrorMessage("addDataError", currentProcessTitle + ":" + e.getMessage(), logger, e);
+        }
+    }
+
+    private void overwriteData(List<Process> processes, String script) {
+        String currentProcessTitle = null;
+        try {
+            script = script.replaceFirst("\\s*action:overwriteData\\s+(.*?)[\r\n\\s]*", "$1");
+            OverwriteDataScript overwriteDataScript = new OverwriteDataScript();
+            for (Process process : processes) {
+                currentProcessTitle = process.getTitle();
+                LegacyMetsModsDigitalDocumentHelper metadataFile = ServiceManager.getProcessService()
+                        .readMetadataFile(process);
+                overwriteDataScript.process(metadataFile, process, script);
+                ServiceManager.getMetsService().saveWorkpiece(metadataFile.getWorkpiece(),
+                        ServiceManager.getProcessService().getMetadataFileUri(process));
+                Helper.setMessage("overwriteDataOk", currentProcessTitle);
+            }
+        } catch (IOException e) {
+            Helper.setErrorMessage("overwriteDataError", currentProcessTitle + ":" + e.getMessage(), logger, e);
```

Comment:
I am not sure if this will work. AFAIK only CDI managed beans have access to the FacesContext that is needed to render messages in the frontend pages. (same goes for the other usages of `Helper.setMessage` or `Helper.setErrorMessage` in this class.)

Reply:
It is actually working!